lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -273,10 +286,21 @@ class AwsCompileStreamEvents {
 
             if (event.stream.consumer && streamType === 'kinesis') {
               if (event.stream.consumer === true) {
-                const consumerName = this.provider.naming.getStreamConsumerName(
+                let consumerName = this.provider.naming.getStreamConsumerName(
                   functionName,
                   streamName
                 );
+                const providerKinesisConfig = this.serverless.service.provider.kinesis;
+                if (
+                  providerKinesisConfig &&
+                  providerKinesisConfig.consumerNamingMode &&
+                  providerKinesisConfig.consumerNamingMode === 'serviceSpecific'
```

Comment:
It would be good to cover these two cases with separate tests

Reply:
We should also add information about that property to documentation about Kinesis

Reply:
> It would be good to cover these two cases with separate tests

@pgrzesik can you give a hint on how to create a test? I'm trying to make use of `runServerless` command when adding a test like this:

```javascript

describe('Naming Mode Deprecation', () => {
  it('should show deprecation message when old naming is used', async () => {
    await runServerless({
      fixture: 'function',
      configExt: {
        functions: {
          foo: {
            events: [{
              stream: {
                type: 'kinesis',
                arn: 'arn:aws:kinesis:us-east-1:027546143534:stream/Foo',
                consumer: true
              },
            }],
          },
        },
      },
      command: 'package',
    });
    // TODO: assert that stdoutData contains deprecation message
  });
});
```

when I run this test it fails with this error: 

```
% npx mocha /Users/petrreshetin/Code/serverless-serverless/test/unit/lib/plugins/aws/package/compile/events/stream.test.js --grep Deprecation                     


  Naming Mode Deprecation
    1) should show deprecation message when old naming is used
2021-09-06T13:21:10.722Z ✖ mocha test fail ServerlessError: consumerNamingMode will be different in v3. The new naming will cause redeployment which may lead to delay in data processing\n' +
  '  More Info: https://www.serverless.com/framework/docs/deprecations/#consumerNamingMode\n' +
  '    at module.exports (lib/utils/logDeprecation.js:88:15)\n' +
```

I tried to wrap this test with expect(...).to.throw(ServerlessError) however it does not work

Reply:
You can ignore deprecations with `runServerless` runs with `disabledDeprecations` passed into `configExt` so it's still possible to test old logic. As for testing if deprecation is displayed, I think it can be skipped

Reply:
> It would be good to cover these two cases with separate tests

Since there is already a test conving standard naming case I just added another one handling the new naming mode case